src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -235,10 +240,26 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
             response.Capabilities.Add("RpcHttpTriggerMetadataRemoved", bool.TrueString);
             response.Capabilities.Add("UseNullableValueDictionaryForHttp", bool.TrueString);
             response.Capabilities.Add("TypedDataCollection", bool.TrueString);
+            response.Capabilities.Add("HandlesWorkerTerminateMessage", bool.TrueString);
 
             return response;
         }
 
+        internal static void WorkerTerminateRequestHandler(WorkerTerminate request)
+        {
+            // Terminate the worker process
+            try
+            {
+                var workerInformation = WorkerInformation.Instance;
+                var workerProcess = Process.GetProcessById(workerInformation.ProcessId);
```

Comment:
Isn't it almost same as what Host was doing? (Process.Kill). I was under the impression that we want to use `IHostApplicationLifetime.StopApplication` method. I know this will internally invoke the other registered callbacks (ex: the event hooked up to `IHostApplicationLifetime.ApplicationStopping` cancellation token.